src/cli/utils/sharepoint.go

```diff
@@ -179,21 +177,10 @@ func IncludeSharePointRestoreDataSelectors(ctx context.Context, opts SharePointO
 		}
 	}
 
-	if slp+sli > 0 {
-		if sli == 0 {
-			opts.ListItem = selectors.Any()
-		}
-
-		opts.ListFolder = trimFolderSlash(opts.ListFolder)
-		containsFolders, prefixFolders := splitFoldersIntoContainsAndPrefix(opts.ListFolder)
-
-		if len(containsFolders) > 0 {
-			sel.Include(sel.ListItems(containsFolders, opts.ListItem))
-		}
-
-		if len(prefixFolders) > 0 {
-			sel.Include(sel.ListItems(prefixFolders, opts.ListItem, selectors.PrefixMatch()))
-		}
+	if sl > 0 {
+		opts.Lists = trimFolderSlash(opts.Lists)
```

Comment:
This could be moved to a common helper which can be used by both groups and sharepoint.

Reply:
@pandeyabs 
we connot generalise this I guess (w/o using generics):
- `opts` type is different for groups (`GroupsOpts`) and sharepoint (`SharePointOpts`)
- `sel` type is different for groups (`GroupsRestore`) and sharepoint (`SharePointRestore`)

Reply:
Thanks! Doesn't seem worthwhile generalizing then.

Reply:
@pandeyabs Had to make them common along with validation of common time flags as well. As I was running into SonarQube duolication fails.

Reply:
yeah, but SonarQube does not agree :)

Reply:
is there any way to not let SonarQube fail for this duplication?

Reply:
It's okay to ignore sonar failure here. We try not to duplicate prod code but sometimes it can happen. This looks fine to me.